### PR TITLE
Introduces A11yTests class to share resources better

### PR DESCRIPTION
**Pull request by felipeleusin**
## Description

I was investigating some slowness when running accessibility tests in Email for instance https://buildkite.com/shopify/email/builds/45509 it seemed strange that those tests would take so long. After investigating a bit I realized that we were opening a lot of browser instances and creating new pages when it wasn't necessary.

Due to the nature of exporting functions with no state it was hard to fix this, I could have a local browser variable for the module but it was strange.

This approach introduces a single class that helds its own browser and pages and performs clean. I also removed some dependency on process.argv and process.exit since this should not be the responsibility of the library and from looking where it was used it is already wrapped in a try/catch anyway.

Let me know what you all think, I opened a draft PR using this in Email (https://github.com/Shopify/email/pull/11832/files). 

If this is something desired I can update the docs/changelog and also web.

PR using the beta version in web: https://github.com/Shopify/web/pull/58844 and https://buildkite.com/shopify/web-storybook-builder/builds/29374
PR using the beta version in email: https://github.com/Shopify/email/pull/11832

From this it looks in web it reduced around 20% the time in web and in email almost 50% reduction for the a11y tests.

## Type of change

- [ ] storybook-a11-test Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have added a changelog entry, prefixed by the type of change noted above (Documentation fix and Test update does not need a changelog as we do not publish new version)

### CODE REVIEW

**Review by BPScott — COMMENTED** [review]
I really like the idea of "launch a browser once and open pages within it" rather than "spawn a new browser for every page" That seems like a huge performance win.

A few things to do before this is ready:

- Can you please fix the linting issues, so that CI passes.
- Can you update the usage instructions in the storybook-a11y-test readme
- Add an entry to storybook-a11y-test's changelog
- Currently we've got no tests for storybook-a11y-test. We'll need to cut a beta, and show this working inside web / polaris-react / a repo of your choosing. If you want to add some regression tests then that'd be amazing but I'm not sure what shape they should take off the top of my head

File: packages/storybook-a11y-test/src/index.ts [hunk 1]
```diff
@@ -28,32 +28,167 @@ declare global {
...
 const FORMATTING_SPACER = '    ';
 
-function getBrowser() {
-  return puppeteer.launch({
-    args: ['--no-sandbox', '--disable-setuid-sandbox'],
-  });
-}
+export class A11yTests {
+  #browser: Browser | undefined;
+  #iframePage: Page | undefined;
```
[the remark this answers is not in the document]
Comment (felipeleusin) [reply packages/storybook-a11y-test/src/index.ts:33]: Yeah but this will fail if used with something that is not the iframe since for the test pages we expect that path + id. Not too confident on this name as well.

**Comment by felipeleusin** [comment]
> I really like the idea of "launch a browser once and open pages within it" rather than "spawn a new browser for every page" That seems like a huge performance win.
> 
> A few things to do before this is ready:
> 
> * Can you please fix the linting issues, so that CI passes.
> * Can you update the usage instructions in the storybook-a11y-test readme
> * Add an entry to storybook-a11y-test's changelog
> * Currently we've got no tests for storybook-a11y-test. We'll need to cut a beta, and show this working inside web / polaris-react / a repo of your choosing. If you want to add some regression tests then that'd be amazing but I'm not sure what shape they should take off the top of my head

Great, will work on this and figure out how to ship a pre-release version to test it out.

**Comment by felipeleusin** [comment]
Got two PRs using a beta version to validate:

PR using the beta version in web: https://github.com/Shopify/web/pull/58844 and https://buildkite.com/shopify/web-storybook-builder/builds/29374
PR using the beta version in email: https://github.com/Shopify/email/pull/11832

From this it looks in web it reduced around 20% the time in web and in email almost 50% reduction for the a11y tests.

**Review by BPScott — APPROVED** [review]
Code changes are looking great to me!

Thanks for updating the READMEs and spinning up a beta to demo.

Two little notes inline, once they're good you should be good to merge

- A thought on an alternative name for currentStoryIds
- We need add a catch clause in the example to handle errors

By the time you come back of holiday we'll probably have released a new version of quilt, so when you merge main, you'll want to accept the newer version numbers from the main branch.

File: packages/storybook-a11y-test/README.md [hunk 2]
```diff
@@ -49,36 +49,39 @@ For optimal test performance, break the build and accessibility testing steps in
...
-    console.error(`‼️  Accessibility violations found`);
-    console.log(results.join('\n'));
-    process.exit(1);
-  } else {
-    console.log('🧚  Accessibility tests passed');
+  const tests = new A11yTests(iframePath);
+
+  try {
+    // Grab all Story IDs
+    const storyIds = await tests.currentStoryIds({
```
Comment (itwasmattgregg) [inline packages/storybook-a11y-test/README.md:62]: Story IDs will be able to be accessed via the json file exported at build time in newer versions of storybook. Eventually we won't need this. But that can be a separate PR. I just think it's worth mentioning here in case that changes your design.
